FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift

```diff
@@ -12,162 +12,179 @@
+  @Environment(\.dismiss) private var dismiss
+  @State private var multiFactorResolver: MultiFactorResolver? = nil
+  @State private var onetimePasscode = ""
+  @State private var showingAlert = false
+
+  @State private var email: String = ""
+  @State private var password: String = ""
+
+  var body: some View {
+    Group {
+      VStack {
+        Group {
+          HStack {
+            VStack {
+              Text("Email/Password Auth")
+                .font(.title)
+                .bold()
+            }
+            Spacer()
+          }
+          HStack {
+            Text(
+              "Login or create an account using the Email/Password auth " +
+                "provider.\n\nEnsure that the Email/Password provider is " +
+                "enabled on the Firebase console for the given project."
+            )
+            .fixedSize(horizontal: false, vertical: true)
+            Spacer()
+          }
+        }
+        .padding(.vertical)
+
+        Spacer()
+        TextField("Email", text: $email)
+          .textFieldStyle(SymbolTextField(symbolName: "person.crop.circle"))
+        TextField("Password", text: $password)
+          .textFieldStyle(SymbolTextField(symbolName: "lock.fill"))
+        Spacer()
+        Group {
+          LoginViewButton(
+            text: "Login",
+            accentColor: .white,
+            backgroundColor: .orange
+          ) {
+            Task {
+              do {
+                _ = try await AppManager.shared
+                  .auth()
+                  .signIn(withEmail: email, password: password)
+//              } catch let error as AuthErrorCode
+//                where error.code == .secondFactorRequired {
+//                // error as? AuthErrorCode == nil because AuthErrorUtils returns generic Errors
+//                // https://firebase.google.com/docs/auth/ios/totp-mfa#sign_in_users_with_a_second_factor
+//                // TODO(ncooke3): Fix?
+              } catch let error as NSError
+                where error.code == AuthErrorCode.secondFactorRequired.rawValue {
+                let mfaKey = AuthErrorUserInfoMultiFactorResolverKey
+                guard
+                  let resolver = error.userInfo[mfaKey] as? MultiFactorResolver,
+                  let multiFactorInfo = resolver.hints.first
+                else { return }
+                if multiFactorInfo.factorID == TOTPMultiFactorID {
+                  // Show the alert to enter the TOTP verification code.
+                  multiFactorResolver = resolver
+                  showingAlert = true
+                } else {
+                  // TODO(ncooke3): Implement handling of other MFA provider (phone).
+                }
+              } catch {
+                print(error.localizedDescription)
+              }
+            }
+          }
+          .alert("Enter one time passcode.", isPresented: $showingAlert) {
+            TextField("Verification Code", text: $onetimePasscode)
```

Comment:
To keep the view's `body` readable, this should probably be a standalone view or a view builder function on the surrounding view

Reply:
Hmm, sorry could you explain this a little bit more? I'm not fully following.